driver-lambda/src/main/com/mongodb/lambdatest/LambdaTestApp.java

```diff
@@ -0,0 +1,143 @@
+                .addCommandListener(new CommandListener() {
+                    @Override
+                    public void commandSucceeded(@NonNull final CommandSucceededEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void commandFailed(@NonNull final CommandFailedEvent event) {
+                        totalCommandCount++;
+                        totalCommandDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                })
+                .applyToServerSettings(builder -> builder.addServerMonitorListener(new ServerMonitorListener() {
+                    @Override
+                    public void serverHeartbeatSucceeded(@NonNull final ServerHeartbeatSucceededEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                    @Override
+                    public void serverHeartbeatFailed(@NonNull final ServerHeartbeatFailedEvent event) {
+                        totalHeartbeatCount++;
+                        totalHeartbeatDuration += event.getElapsedTime(MILLISECONDS);
+                    }
+                }))
+                .applyToConnectionPoolSettings(builder -> builder.addConnectionPoolListener(new ConnectionPoolListener() {
+                    @Override
+                    public void connectionCreated(@NonNull final ConnectionCreatedEvent event) {
+                        openConnections++;
+                    }
+                    @Override
+                    public void connectionClosed(@NonNull final ConnectionClosedEvent event) {
+                        openConnections--;
+                    }
+                }))
+                .build();
+        mongoClient = MongoClients.create(settings);
+    }
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayProxyRequestEvent input, final Context context) {
+        try {
+            MongoCollection<Document> collection = mongoClient
+                    .getDatabase("lambdaTest")
+                    .getCollection("test");
+            BsonValue id = collection.insertOne(new Document("n", 1)).getInsertedId();
+            collection.deleteOne(new Document("_id", id));
+
+            BsonDocument responseBody = new BsonDocument()
+                    .append("totalCommandDuration", new BsonDouble(totalCommandDuration))
+                    .append("totalCommandCount", new BsonDouble(totalCommandCount))
+                    .append("totalHeartbeatDuration", new BsonDouble(totalHeartbeatDuration))
+                    .append("totalHeartbeatCount", new BsonDouble(totalHeartbeatCount))
+                    .append("openConnections", new BsonDouble(openConnections));
+
+            return templateResponse()
```

Comment:
I do not believe capturing a running total violates the quoted spec.

We took a similar "running total" approach in the Kafka JMX. I believe that the user is able to derive any timings, by taking a sample S when a reset would otherwise happen, and then subtracting S from the usual sample E, to get the change during the interval. The benefit is that the user has precise control over when the "reset" happens (that is, the start point), and queries do not change the application state. There are, of course, circumstances where a "hardcoded" start and end must be used (for example, the execution of a particular process must be timed very rigorously), but I do not think there is such a requirement here.

(I hope I am not misreading what was meant by separate tracking of commands/heartbeats - there is both `totalCommandDuration`, and `totalHeartbeatDuration`.)

Reply:
Thank you for providing this explanation. As I understand it, the testing aims to highlight the unique differences between lambda invocations. Maintaining separate results for each invocation ensures clear and distinct data in the logs without the need for additional calculations. For those interested in cumulative totals, they can perform the necessary calculations using the recorded durations.

However, I don't have a strong preference here, so feel free to resolve this thread

Reply:
I checked with `@durran` and the resetting behaviour was indeed preferred, so I have changed it (with the exception of not resetting the open connection count, which should not be reset).